packages/swap/src/routes/v2/quote.ts

```diff
@@ -0,0 +1,334 @@
+import { Asset, assetConstants, Assets, Chain, Chains, InternalAsset } from '@/shared/enums';
+import { quoteQuerySchema, QuoteQueryResponse } from '@/shared/schemas';
+import env from '../../config/env';
+import { getBoostSafeMode } from '../../polkadot/api';
+import { getUsdValue } from '../../pricing/checkPriceWarning';
+import Quoter from '../../quoting/Quoter';
+import { getBoostFeeBpsForAmount } from '../../utils/boost';
+import getPoolQuote from '../../utils/getPoolQuote';
+import logger from '../../utils/logger';
+import { getPools, getTotalLiquidity } from '../../utils/pools';
+import { getIngressFee, validateSwapAmount } from '../../utils/rpc';
+import ServiceError from '../../utils/ServiceError';
+import { asyncHandler } from '../common';
+
+type AdditionalInfo = {
+  srcAsset: InternalAsset;
+  destAsset: InternalAsset;
+  amount: string;
+  usdValue: string | undefined;
+  limitOrdersReceived: Awaited<ReturnType<Quoter['getLimitOrders']>> | undefined;
+};
+
+const handleQuotingError = (err: unknown, info: AdditionalInfo) => {
+  if (err instanceof ServiceError) throw err;
+
+  const message = err instanceof Error ? err.message : 'unknown error (possibly no liquidity)';
+
+  if (message.includes('InsufficientLiquidity')) {
+    logger.info('insufficient liquidity received', info);
+    throw ServiceError.badRequest('insufficient liquidity for requested amount');
+  }
+
+  logger.error('error while collecting quotes:', err);
+
+  return message;
+};
+
+export const getDcaQuoteParams = async (asset: InternalAsset, amount: bigint) => {
+  const usdChunkSize =
+    env.DCA_USD_CHUNK_SIZE?.find((item) => item.asset === asset)?.usdChunkSize ?? 3000;
+
+  const usdValue = await getUsdValue(amount, asset).catch(() => undefined);
+  if (!usdValue) {
+    logger.error(
+      `could not get usd value for DCA quote calculation. asset: ${asset} , amount: ${amount}`,
+    );
+    return null;
+  }
+  if (Number(usdValue) < usdChunkSize) {
+    return {
+      chunkSize: amount,
+      lastChunkAmount: amount,
+      numberOfChunks: 1,
+      addedDurationSeconds: 0,
+    };
+  }
+  const numberOfChunksPrecise = Number(usdValue) / usdChunkSize;
+  const chunkSize = BigNumber(amount.toString()).dividedBy(numberOfChunksPrecise);
+
+  const numberOfChunks =
+    numberOfChunksPrecise - Math.floor(numberOfChunksPrecise) < 0.05
+      ? Math.floor(numberOfChunksPrecise)
+      : Math.ceil(numberOfChunksPrecise);
+
+  const lastChunkAmount = BigInt(
+    new BigNumber(amount.toString()).minus(chunkSize.multipliedBy(numberOfChunks - 1)).toFixed(0),
+  );
+
+  return {
+    chunkSize: BigInt(chunkSize.toFixed(0)),
+    lastChunkAmount,
+    numberOfChunks,
+    addedDurationSeconds: Math.ceil(env.DCA_CHUNK_INTERVAL_SECONDS * (numberOfChunks - 1)), // we deduct 1 chunk because the first one is already accounted for in the regular quote
+  };
+};
+
+const adjustDcaQuote = ({
+  dcaQuoteParams,
+  dcaQuote,
+  dcaBoostedQuote,
+  estimatedBoostFeeBps,
```

Comment:
not sure if we need `estimatedBoostFeeBps` - cant we just scale the boost fee if it is there and ignore if it is not?

Reply:
`dcaBoostedQuote` doesn't already contain `estimatedBoostFeeBps`. So when we're populating `dcaQuote.boostQuote`, we need to also use this value.

Reply:
okay i see - i think we should set the `estimatedBoostFeeBps` in `getPoolQuote` then 🙂 it would make the existing code a bit easier and that method already adds the fee to the fee array, so it already has all the data it needs